From edc1c3cae8a090eb0d8a132d9c4b28df47d2f6cc Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@archlinux.org>
Date: Sun, 3 Sep 2017 03:53:39 -0400
Subject: [PATCH 8/9] makechrootpkg: Fix unconditionally running namcap

Fixes regression in 2fd5931a8c67289a8a4acd327b3ce99a5d64c8c7

$run_namcap will always be set to ""
`if $not_a_var; then ...; fi` is always truthful when $not_a_var is
unset or equal to "" and the `then` clause will always be run.

I'm not sure why global state variables need to be cloned locally for
their sole explicit purpose.

But for now this patch implements the minimum necessary work to properly
pass the "do I want namcap" variable into prepare_chroot() according to
the current logic flow.
Note that I have still not thorougly tested makechrootpkg.

Signed-off-by: Eli Schwartz <eschwartz@archlinux.org>
---
 makechrootpkg.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makechrootpkg.in b/makechrootpkg.in
index 6b53337..d6f48b2 100644
--- a/makechrootpkg.in
+++ b/makechrootpkg.in
@@ -417,7 +417,7 @@ main() {
 
 	download_sources "$copydir" "$makepkg_user"
 
-	prepare_chroot "$copydir" "$USER_HOME" "$keepbuilddir"
+	prepare_chroot "$copydir" "$USER_HOME" "$keepbuilddir" "$run_namcap"
 
 	if arch-nspawn $cache_dir "$copydir" \
 		--bind="$PWD:/startdir" \
-- 
2.16.2

